### ISSUE TITLE
very confused about the test method in finetune.py

### ISSUE THREAD

**Issue by passion3394** [JiaRenChang/PSMNet#80]
In the following code segment:
## training ##
        for batch_idx, (imgL_crop, imgR_crop, disp_crop_L) in enumerate(TrainImgLoader):
            start_time = time.time()

            loss = train(imgL_crop,imgR_crop, disp_crop_L)
            print('Iter %d training loss = %.3f , time = %.2f' %(batch_idx, loss, time.time() - start_time))
            total_train_loss += loss
            print('epoch %d total training loss = %.3f' %(epoch, total_train_loss/len(TrainImgLoader)))

            ## Test ##

            for batch_idx, (imgL, imgR, disp_L) in enumerate(TestImgLoader):
                test_loss = test(imgL,imgR, disp_L)
                print('Iter %d 3-px error in val = %.3f' %(batch_idx, test_loss*100))
                total_test_loss += test_loss

            print('epoch %d total 3-px error in val = %.3f' %(epoch, total_test_loss/len(TestImgLoader)*100))
            if total_test_loss/len(TestImgLoader)*100 > max_acc:

I found that after every batch of training, the test on the validation set is performed, this way is very time consuming. Let me calculate the time, If both the batch_size of training and testing are 2, the training time is about 1.8s and the testing size is about 30s, then the time of every epoch is 80*(1.8+30)=2544seconds=42min, It's too long. After the training of 300 epochs, 8 days have been gone.
What's your consideration about the above code?

**Comment by JiaRenChang** [comment]
Hi @passion3394 ,
We validate every epoch for selecting best model.
On our 4 titan Xs machine, it takes around 4 hours for 300 epochs.

**Comment by passion3394** [comment]
@JiaRenChang I think the code validate every iter, not every epoch, is that right?

**Comment by passion3394** [comment]
@JiaRenChang sorry, you are right. I took some mistake around the tab format of the code.
